client/src/components/Query/Filter.jsx

```diff
@@ -0,0 +1,355 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions, no-case-declarations */
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Row, Col, Typography, Card, Tag, Input, Popover, Dropdown, Button, Radio, Icon, Checkbox,
+} from 'antd';
+
+import IconKit from 'react-icons-kit';
+import {
+  empty, one, full, info,
+} from 'react-icons-kit/entypo';
+
+// import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
+
+
+export const FILTER_TYPE_GENERIC = 'generic';
+export const FILTER_TYPE_SPECIFIC = 'specific';
+export const FILTER_OPERATOR_TYPE_ALL = 'all';
+export const FILTER_OPERATOR_TYPE_ONE = 'one';
+export const FILTER_OPERATOR_TYPE_NONE = 'none';
+
+const createPopoverByFilterType = (state) => {
+  const { data } = state;
+  const { type, operator } = data;
+  let content = null;
+  let legend = null;
+
+  switch (type) {
+    case FILTER_TYPE_GENERIC:
+      content = (
+        <div>
+          <Typography.Text>FILTER is OPERATOR</Typography.Text>
+          <ul>
+            <li>VALUE 1</li>
+            <li>VALUE 3</li>
+          </ul>
+        </div>
+      );
+      switch (operator) {
+        default:
+        case FILTER_OPERATOR_TYPE_ALL:
+          legend = (<IconKit size={16} icon={full} />);
+          break;
+        case FILTER_OPERATOR_TYPE_ONE:
+          legend = (<IconKit size={16} icon={one} />);
+          break;
+        case FILTER_OPERATOR_TYPE_NONE:
+          legend = (<IconKit size={16} icon={empty} />);
+          break;
+      }
+      break;
+
+    case FILTER_TYPE_SPECIFIC:
+    default:
+      legend = (<IconKit size={16} icon={info} />);
+      break;
+  }
+
+  return (
+    <Popover
+      className="legend"
+      trigger="hover"
+      placement="topLeft"
+      content={content}
+    >
+      {legend}
+    </Popover>
+  );
+};
+
+class Filter extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      type: null,
+      options: {
+        editable: null,
+        selectable: null,
+      },
+      data: {},
+      draft: {},
+      visible: null,
+      selected: null,
+      opened: null,
+    };
+    this.isEditable = this.isEditable.bind(this);
```

Comment:
I suggest you use a tool like `auto-bind-es5` instead:

```
import autobind from 'auto-bind-es5';
...
class BananaComponent extends Component {
  constructor(props) {
    super(props);
    autobind(this);
  }
```

Reply:
@jberube I could, but I want to keep dependencies to a minimal. :)